mypy/main.py

```diff
@@ -171,14 +171,69 @@ def process_options(args: List[str]) -> Tuple[List[BuildSource], Options]:
     if not args:
         usage('Missing target file or module')
 
-    if args[1:]:
-        usage('Extra argument: {}'.format(args[1]))
-
     if options.python_path and options.pyversion[0] == 2:
         usage('Python version 2 (or --py2) specified, '
               'but --use-python-path will search in sys.path of Python 3')
 
-    return [BuildSource(args[0], None, None)], options
+    targets = []
+    for arg in args:
+        if arg.endswith('.py'):
```

Comment:
But if you want me to I can treat .py and .pyi the same in all the code I added.